include/qpid/dispatch/router_core.h

```diff
@@ -797,4 +799,9 @@ qdr_connection_info_t *qdr_connection_info(bool             is_encrypted,
                                            int              ssl_ssf,
                                            bool             ssl);
 
+typedef struct qdr_timer_work_t qdr_timer_work_t;
+typedef void (*qdr_timer_cb_t)(qdr_core_t *core, void* context);
+qdr_timer_work_t *qdr_timer_schedule(qdr_core_t *core, qdr_timer_cb_t callback, void *timer_context, int timer_delay);
+void qdr_timer_delete(qdr_core_t *core, qdr_timer_work_t *timer_work);
```

Comment:
Some comments on these function prototypes would be helpful?  What exactly does qdr_timer_delete do?  Should there be a qdr_timer_cancel()?

Reply:
Other timer APIs we've created look more like this:

qdr_core_timer_t *qdr_core_timer();
void qdr_core_timer_schedule(qdr_core_timer_t *timer, ...);
void qdr_core_timer_cancel(qdr_core_timer_t *timer);
void qdr_core_timer_free(qdr_core_timer_t *timer);

This prevents you from having to create/allocate a new timer each time you reschedule.  The timers we will be using will be recurring, so this more classic API fits well.